packages/fx-core/src/component/debugHandler/sso.ts

```diff
@@ -0,0 +1,230 @@
+  }
+
+  // TODO: output message
+  public async setUp(): Promise<Result<Void, FxError>> {
+    try {
+      const checkArgsResult = await this.checkArgs();
+      if (checkArgsResult.isErr()) {
+        return err(checkArgsResult.error);
+      }
+
+      const projectSettingsResult = await loadProjectSettingsByProjectPath(this.projectPath, true);
+      if (projectSettingsResult.isErr()) {
+        return err(projectSettingsResult.error);
+      }
+
+      const projectSettingsV3: ProjectSettingsV3 = projectSettingsResult.value as ProjectSettingsV3;
+      const cryptoProvider = new LocalCrypto(projectSettingsV3.projectId);
+
+      const envInfoResult = await environmentManager.loadEnvInfo(
+        this.projectPath,
+        cryptoProvider,
+        environmentManager.getLocalEnvName(),
+        true
+      );
+      if (envInfoResult.isErr()) {
+        return err(envInfoResult.error);
+      }
+
+      const envInfoV3: v3.EnvInfoV3 = envInfoResult.value;
+      envInfoV3.state[ComponentNames.AadApp] = envInfoV3.state[ComponentNames.AadApp] || {};
+
+      // set objectId, clientId, clientSecret, oauth2PermissionScopeId from args to state
+      if (checkArgsResult.value) {
+        envInfoV3.state[ComponentNames.AadApp].objectId = this.args.objectId;
+        envInfoV3.state[ComponentNames.AadApp].clientId = this.args.clientId;
+        envInfoV3.state[ComponentNames.AadApp].clientSecret = this.args.clientSecret;
+        envInfoV3.state[ComponentNames.AadApp].oauth2PermissionScopeId =
+          this.args.accessAsUserScopeId || uuidv4();
+      }
+
+      TokenProvider.init({
+        m365: this.m365TokenProvider,
+      });
+
+      // set oauth2PermissionScopeId to state
+      envInfoV3.state[ComponentNames.AadApp].oauth2PermissionScopeId =
+        envInfoV3.state[ComponentNames.AadApp].oauth2PermissionScopeId || uuidv4();
+
+      // not using exsting AAD app and not yet created
+      if (!envInfoV3.state[ComponentNames.AadApp].objectId) {
+        const context = this.constructPluginContext(envInfoV3, cryptoProvider);
+        const manifest = await AadAppManifestManager.loadAadManifest(context);
+
+        const config = new ProvisionConfig(true, false);
+        await AadAppClient.createAadAppUsingManifest(
+          TelemetryEvent.DebugSetUpSSO,
+          manifest,
+          config
+        );
+        await AadAppClient.createAadAppSecret(TelemetryEvent.DebugSetUpSSO, config);
+
+        // set objectId, clientId, clientSecret to state
+        envInfoV3.state[ComponentNames.AadApp].objectId = config.objectId;
+        envInfoV3.state[ComponentNames.AadApp].clientId = config.clientId;
+        envInfoV3.state[ComponentNames.AadApp].clientSecret = config.password;
+      }
+
+      // set applicationIdUris to state
+      let applicationIdUri = "api://";
+      if (ProjectSettingsHelper.includeFrontend(projectSettingsV3)) {
+        applicationIdUri += `localhost/${envInfoV3.state[ComponentNames.AadApp].clientId}`;
```

Comment:
For tab + bot scenario, app id uri should be "api://localhost/botid-xxx", client id should not be included?

Reply:
Good catch! Fixed, thanks.